reference/libraries/3d/gpu-debugging.md

```diff
@@ -0,0 +1,124 @@
+
+- Focus on any Stride window (e.g., a render window or scene window).
+- Press **F4** to enable the Stride Profiler.
+
+The Stride Profiler will appear as a small text line in the top left corner of the window, showing FPS and other information. It has three states: FPS, CPU page, and GPU page. It doesn’t matter which page is currently displayed, but the profiler **must be enabled** for RenderDoc to receive the necessary information for organizing GPU calls.
+
+##### 3. Capturing Frames
+
+- **Single Frame Capture**: Trigger the **Capture Next Frame** input to capture the next frame rendered by the GPU. This frame is stored for later analysis in RenderDoc.
+  
+- **Multiple Frame Captures**: Use the **Number of Frames to Capture** input to capture a series of successive frames. This is particularly useful when debugging recurring issues like periodic frame drops or performance dips. For example, if a problem occurs every `n` frames, you can set the number of frames to `n` and capture the relevant frames for analysis.
+
+![](../../../images/libraries/3d/renderdoc_node.png)
+
+### Attaching RenderDoc to vvvv
+
+1. Start RenderDoc.
+2. Go to **File** > **Attach to Running Instance**.
+3. In the pop-up window, look for **localhost** and find **vvvv** underneath it.
+4. Double-click **vvvv** to connect RenderDoc to the vvvv instance and access the captured frames. If no frames have been captured yet, nothing will appear, but you can trigger frame capture from vvvv or directly in RenderDoc.
+
+![](../../../images/libraries/3d/attach.png)
+
+### Debugging Captured Frames in RenderDoc
+
+Once frames are captured and loaded in RenderDoc:
+1. **Double-click** on a captured frame:
+   - RenderDoc will analyze the frame, which may take a few seconds.
+   - You will see all GPU calls made by vvvv during that frame.
+
+2. **Timeline**: At the top of the RenderDoc interface, a timeline displays the sequence of GPU events.
+
+3. **Draw Calls and Pipeline State**: Below the timeline, you’ll find the draw calls and the pipeline state, where you can inspect shaders, resources, and other rendering details.
+
+4. **Shader Debugging**: 
+   - To debug pixel, vertex, or compute shaders, go to the **Pipeline State** section.
+   - For more information on debugging shaders in RenderDoc, refer to their official guide: [Shader Debugging in RenderDoc](https://renderdoc.org/docs/how/how_debug_shader.html).
+
+![](../../../images/libraries/3d/renderdoc.png)
+
+### Performance Profiling with RenderDoc
+
+RenderDoc also provides performance counters that help assess GPU performance metrics for specific frames. These counters give you deeper insights into which GPU calls are the most performance-heavy.
+
+#### 1. Enabling the Performance Counter Viewer
+
+- Go to **Window > Performance Counter Viewer** in RenderDoc.
+- In this view, you can access different GPU counters, including generic counters and **Nvidia-specific counters** (if you install an additional DLL).
```

Comment:
"install an additional DLL" is that something that can be elaborated upon?

Reply:
Yes, this is a message that gets shown in RenderDoc like this:

<img width="346" alt="2024-10-23 10_26_09-Performance counter selection" src="https://github.com/user-attachments/assets/7bb66284-0907-4b4d-91fc-e33add404385">

But I've added a section explaining the installation.